### PR TITLE
feat(logger)add log service

### PR DESCRIPTION
**Pull request by yuvalbl**
should solve issue #947

### CODE REVIEW

File: src/app/packages/core/utils/ansyn-logger.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		if (includeBrowserData) {
+			prefix += `[window:${window.innerWidth}x${window.innerHeight}][userAgent: ${navigator.userAgent}]`
+		}
+		const str = `${prefix}[${severity}] ${msg}`;
+		this._output(severity, str);
+	}
+
+	private _output(severity: Severity, msg: string) {
+		switch (severity) {
+			// case 'CRITICAL':
```
Comment (yuvalbl) [inline src/app/packages/core/utils/ansyn-logger.ts]: This is commented accorting to @asafMasa request

File: src/app/packages/core/utils/ansyn-logger.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	warn(msg: string);
+	info(msg: string);
+}
+
+export class AnsynLogger implements MinimalLogger {
+	static activeLogger = new AnsynLogger('ENV');	// default logger (unknown environment)
+	env: string;
+	standardPrefix: string;
+
+	constructor(environment: string) {
```
Comment (AmitMY) [inline src/app/packages/core/utils/ansyn-logger.ts]: public env: string

File: src/app/packages/core/utils/ansyn-logger.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		this._log('WARNING', msg);
+	}
+	info(msg: string) {
+		this._log('INFO', msg);
+	}
+	debug(msg: string) {
+		this._log('DEBUG', msg);
+	}
+
+	private _log(severity: Severity, msg: string, includeBrowserData?: boolean) {
```
Comment (AmitMY) [inline src/app/packages/core/utils/ansyn-logger.ts]: remove _

protected

File: src/app/packages/core/utils/ansyn-logger.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		let prefix = `${this.standardPrefix}[${Date()}]`;
+		if (includeBrowserData) {
+			prefix += `[window:${window.innerWidth}x${window.innerHeight}][userAgent: ${navigator.userAgent}]`
```
Comment (AmitMY) [inline src/app/packages/core/utils/ansyn-logger.ts]: import version form package

File: src/app/packages/core/utils/ansyn-logger.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+		}
+		const str = `${prefix}[${severity}] ${msg}`;
+		this._output(severity, str);
+	}
+
+	private _output(severity: Severity, msg: string) {
```
Comment (AmitMY) [inline src/app/packages/core/utils/ansyn-logger.ts]: protected

File: src/assets/app-loader/app-loader.js [hunk 6]
```diff
@@ -1,3 +1,5 @@
+import {AnsynLogger} from "../../app/packages/core/utils/ansyn-logger";
```
Comment (AmitMY) [inline src/assets/app-loader/app-loader.js]: ahm no

File: src/app/packages/core/utils/ansyn-logger.ts (hunk 2 above)
Comment (yuvalbl) [reply src/app/packages/core/utils/ansyn-logger.ts]: OK

File: src/app/packages/core/utils/ansyn-logger.ts (hunk 3 above)
Comment (yuvalbl) [reply src/app/packages/core/utils/ansyn-logger.ts]: K

File: src/app/packages/core/utils/ansyn-logger.ts (hunk 4 above)
Comment (yuvalbl) [reply src/app/packages/core/utils/ansyn-logger.ts]: on production package will not be available. let's postpone this one for a later time (it is not part of the basic requirements of this feature)

File: src/app/packages/core/utils/ansyn-logger.ts (hunk 5 above)
Comment (yuvalbl) [reply src/app/packages/core/utils/ansyn-logger.ts]: Now it is

File: src/assets/app-loader/app-loader.js (hunk 6 above)
Comment (yuvalbl) [reply src/assets/app-loader/app-loader.js]: fixed

File: src/app/app/app-providers/fetch-config-providers.ts [hunk 7]
```diff
@@ -69,6 +70,10 @@ export const getProviders = (conf): any[] => {
 
 export const fetchConfigProviders = fetch('/assets/config/app.config.json')
 	.then(response => response.json())
+	.then(conf => {
+		AnsynLogger.activeLogger = new AnsynLogger(conf.loggerConfig.env);
```
Comment (AmitMY) [inline src/app/app/app-providers/fetch-config-providers.ts]: why different environment for the logger? we already have an environment config

File: src/app/packages/core/utils/ansyn-logger.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+	env: string;
+	standardPrefix: string;
+
+
+
+	constructor(env: string) {
+		this.standardPrefix = `Ansyn[${this.env}]`
+	}
+
+	critical(msg: string) {
```
Comment (AmitMY) [inline src/app/packages/core/utils/ansyn-logger.ts]: you never call `critical`, `error`, `warn`, `info`, and `debug`, only the methods of the static `activeLogger`

Keep the static stuff, they are fine, but move the logging into  new class that implements `MinimalLogger`. Too much redundancy here

Comment (yuvalbl) [reply src/app/packages/core/utils/ansyn-logger.ts]: FAD, see #947

File: src/app/app/app-providers/fetch-config-providers.ts (hunk 7 above)
Comment (yuvalbl) [reply src/app/app/app-providers/fetch-config-providers.ts]: This is according to design.

Comment (yuvalbl) [reply src/app/app/app-providers/fetch-config-providers.ts]: everything is static now

**Pull request merged**